### ISSUE TITLE
Add support for resources

### ISSUE THREAD

**Issue by fniephaus** [hpi-swa/smalltalkCI#190]
Sometimes it is necessary to access external resources in order to be able to load and test a project. Currently, smalltalkCI symlinks the project directory into the build directory and it's possible (but inconvenient) to copy files into the build dir. A better approach would be to have smalltalkCI set the default working directory in an image to the project's directory. However, I am currently not sure if this is possible and if this really solves the problem.

**Comment by SergeStinckwich** [comment]
Yes this is definitively something I would like to use in the future.

**Comment by dalehenrich** [comment]
Agreed, I think that being able to specify the location of the image's working directory is important ... hmmm ... with FileSystem wouldn't it be possible to define a named location ... so perhaps there should/could be a set of named directories defined within the configuration? FileSystem isn't present in GemStone at the moment, but soon it will be (next 6 months or so?) ... anyway, just some more thoughts

**Comment by fniephaus** [comment]
`FileSystem` isn't present in Squeak images, so we'd need to come up with a solution for that. I'm curious, how do these named locations work? I haven't heard of them yet.

**Comment by dalehenrich** [comment]
There is a class FileLocator in FileSystem that has the notion of named origins. Here's a list of supported origins for the Pharo image: `#(image imageDirectory changes vmBinary vmDirectory workingDirectory localDirectory)` ... it's not too big of stretch to imagine adding additional origins for SmalltalkCI or being able to specify a different origin location based on the configuration ...

**Comment by fniephaus** [comment]
That's interesting. I wish something like that would be available in Squeak, but I'm afraid there isn't yet. smalltalkCI could provide a similar facility, but then tests would need to make assumptions about smalltalkCI and I don't think we want tests to depend on this. Maybe we should just change the default working directory for now. By default, we'll use the root dir of the project, and we can add a config slot to specify a custom directory in a second step.

**Comment by dalehenrich** [comment]
I agree that working directory is **required** ...

**Comment by fniephaus** [comment]
This issue comes up every now and then and I thought about it again.
It seems like there's no convenient way to readjust the working directory in Pharo and I haven't had a look at how this would work in GemStone.
Also, I like the idea of separate build directories and I want to avoid touching the original project directory as well.
Having said that, the best way to handle resources seems to be to make it part of the config file and have smalltalkCI copy the files into the build directory (for convenience, but also for consistency).
Example:
```javascript
SmalltalkCISpec {
  #resources : [ 'mydata', 'my/other/resource/directory', 'yetAnotherRequiredFile.st' ],
  #loading : ...
}
```
... and then smalltalkCI would copy `./mydata`, `./my/other/resource/directory`, and `./yetAnotherRequiredFile.st` into the build directory before loading the project.
